templates/500.html

```diff
@@ -14,10 +18,14 @@
             <img src="/static/images/errors/500art.svg" alt=""/>
             <div class="errorbox">
                 <div class="errorcontent">
-                    <h1 class="lead">Internal server error</h1>
-                    <p>This Zulip server is currently experiencing some technical difficulties. Sorry about that!</p>
-                    <p>The page will reload automatically soon after service is restored.</p>
-                    <p>If you'd like, you can <a href="mailto:{{ support_email }}?Subject=500%20error%20on%20%7Bwhich%20URL%3F%7D&Body=Hi%20there%21%0A%0AI%20was%20trying%20to%20do%20%7Bwhat%20were%20you%20trying%20to%20do%3F%7D%20at%20around%20%7Bwhen%20was%20this%3F%7D%20when%20I%20got%20a%20500%20error%20while%20accessing%20%7Bwhich%20URL%3F%7D.%0A%0AThanks!%0A%0ASincerely%2C%20%0A%0A%7BYour%20name%7D">drop us a line</a> to let us know what happened.</p>
+                    <h1 class="lead">{{ _("Internal server error") }}</h1>
+                    <p>{{ _("This Zulip server is currently experiencing some technical difficulties. Sorry about that!") }}</p>
+                    <p>{{ _("The page will reload automatically soon after service is restored.") }}</p>
+                    <p>
+                        {% trans %}
+                        If you'd like, you can <a href="mailto:{{ support_email }}?Subject=500%20error%20on%20%7Bwhich%20URL%3F%7D&Body=Hi%20there%21%0A%0AI%20was%20trying%20to%20do%20%7Bwhat%20were%20you%20trying%20to%20do%3F%7D%20at%20around%20%7Bwhen%20was%20this%3F%7D%20when%20I%20got%20a%20500%20error%20while%20accessing%20%7Bwhich%20URL%3F%7D.%0A%0AThanks!%0A%0ASincerely%2C%20%0A%0A%7BYour%20name%7D">drop us a line</a> to let us know what happened.
+                        {% endtrans %}
+                    </p>
```

Comment:
Noted that this template wasn't translated, so added that as well.

Reply:
Complex HTML tags shouldn’t be written directly in translated strings. Translators will be confused or insert invalid markup.

Reply:
I saw that `unsubscribe_link_error.html` has a similar translation string situation with a support email link, so I thought that wouldn't be an issue. But maybe [that's just been around long enough](https://github.com/zulip/zulip/commit/5359e6b0d4af5d0b9a4180d074252290f49cd5e7#diff-12e2ee50a86a18160897680f0ec40d74b800786143d7c695c0abb5dcf40eeaca) that it the translator issues have been ironed out?

I can set a variable, say `SUPPORT_EMAIL_URL_ENCODING`, in the jinja template and then use that in the HTML anchor tag. It will be enclosed in curly brackets, so the translators should be familiar with that indicating a variable that shouldn't be translated.

```
{% set SUPPORT_EMAIL_URL_ENCODING = "?Subject=500%20error%20on%20%7Bwhich%20URL%3F%7D&Body=Hi%20there%21%0A%0AI%20was%20trying%20to%20do%20%7Bwhat%20were%20you%20trying%20to%20do%3F%7D%20at%20around%20%7Bwhen%20was%20this%3F%7D%20when%20I%20got%20a%20500%20error%20while%20accessing%20%7Bwhich%20URL%3F%7D.%0A%0AThanks!%0A%0ASincerely%2C%20%0A%0A%7BYour%20name%7D" %}

...
    <p>
        {% trans %}
        If you'd like, you can <a href="mailto:{{ support_email }}{{ SUPPORT_EMAIL_URL_ENCODING }}">drop us a line</a> to let us know what happened.
        {% endtrans %}
    </p>
```

And if we prefer that structure, I could update the similar instance in `unsubscribe_link_error.html` as well. Maybe as a follow-up PR update [this section](https://zulip.readthedocs.io/en/latest/translating/internationalization.html?highlight=_#jinja2-templates) in the contributor documentation as well in case it comes up again in the future? I did reference that before going through to do this update.

Reply:
Hmm... nope. The template parser does not expect the URL encoded string.

Reply:
I believe `templates/zerver/emails/missed_message.source.html` has some examples of the nicer pattern we've started using for HTML content inside translated strings in Django templates.

Can you also start a chat.zulip.org thread about potentially changing the prefilled email content in that link? I suspect nobody has thought about it since 2013.

Reply:
> I believe `templates/zerver/emails/missed_message.source.html` has some examples of the nicer pattern we've started using for HTML content inside translated strings in Django templates.

That is much nicer, but would still have the problem, in this instance, of triggering the template parser tool because of `%` being part of the URL encoding. The template parser expects that character to only be the end of the template tag. So, we could rewrite that tool logic for these two cases if we feel like we'd like to be able to hard-code variables like this in templates.

> Can you also start a chat.zulip.org thread about potentially changing the prefilled email content in that link? I suspect nobody has thought about it since 2013.

Sure. Going with #general as the stream for the conversation, but someone can move it if they feel like it would be better placed elsewhere.

Reply:
Resolved via #22926.